sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/v2Commands.scala

```diff
@@ -510,11 +510,21 @@ case class ShowTables(
   override def children: Seq[LogicalPlan] = Seq(namespace)
 }
 
-object ShowTables {
+trait ShowTablesLegacyHelper {
+  def getOutputAttrs: Seq[Attribute]
```

Comment:
Is this a good idea that the code to handle this legacy behaviour depends on `tirait`? If we remove this legacy behaviour in far future and we remove this trait, the change can lead to binary-incompatibility?

Reply:
The callers are bundled in the catalyst module, I guess it's safe?

Reply:
(Sorry for my late reply...) How about simply inlining `getLegacyOutputAttrs`? It seems there are the only two places where  `getLegacyOutputAttrs` is used, so I'm not sure that we need this trait.